internal/agent.go

```diff
@@ -80,38 +79,52 @@ func getAgentID() (string, error) {
 }
 
 // Start the Agent. This will start the discovery ticker and the heartbeat ticker
-func (a *Agent) Start() error {
-	var wg sync.WaitGroup
+func (a *Agent) Start(ctx context.Context) error {
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
shadowing, and reassigning the ctx variable from parameter can lead to some smell, maybe we can rename the ctx of the errogroup to something like groupCtx, and use it in the flow like you did below, so we can distinguish the two contexts

Reply:
Done